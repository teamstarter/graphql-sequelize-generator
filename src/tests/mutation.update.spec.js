const request = require('supertest')
const { deleteTables } = require('./testDatabase.js')
const { createServer, closeServer, resetDb } = require('./setupServer')

/**
 * Starting the tests
 */
describe('Test the create mutation', () => {
  let server = null
  let trace = []
  const globalPreCallback = (type) => {
    trace.push(type)
  }

  beforeAll(async () => {
    server = await createServer({}, globalPreCallback)
  })

  afterAll(() => closeServer(server))

  beforeEach(async () => {
    trace = []
    await resetDb()
  })

  afterEach(async () => {
    await deleteTables()
  })

  it('Check that you can create a new instance of one entity', async () => {
    const responseUser5PreUpdate = await request(server)
      .get(
        `/graphql?query=
        query getUsers {
          user: user(id: 5) {
            id
            name
            company {
              id
              name
            }
          }
        }
        &operationName=getUsers`
      )
      .set('userId', 1)
    expect(responseUser5PreUpdate.body.errors).toBeUndefined()
    expect(responseUser5PreUpdate.body.data.user).not.toBeUndefined()
    expect(responseUser5PreUpdate.body.data.user).toMatchSnapshot(
      'User 5 before update'
    )

    const responseUserCreate = await request(server)
      .post('/graphql')
      .set('userid', 1)
      .send({
        query: `mutation userUpdate($user: userInput!) {
              user : userUpdate(user: $user) {
                id
                name
                __typename
              }
            }`,
        variables: {
          user: {
            id: 5,
            name: 'edited name',
          },
        },
        operationName: 'userUpdate',
      })
    expect(responseUserCreate.body.errors).toBeUndefined()
    expect(responseUserCreate.body.data.user).not.toBeUndefined()
    expect(responseUserCreate.body.data.user).toMatchSnapshot('Updated user')

    const responseUser5PostUpdate = await request(server)
      .get(
        `/graphql?query=
        query getUsers {
          user: user(id: 5) {
            id
            name
            company {
              id
              name
            }
          }
        }
        &operationName=getUsers`
      )
      .set('userId', 1)
    expect(responseUser5PostUpdate.body.errors).toBeUndefined()
    expect(responseUser5PostUpdate.body.data.user).not.toBeUndefined()
    expect(responseUser5PostUpdate.body.data.user).toMatchSnapshot(
      'User 5 post update'
    )
    expect(trace).toMatchSnapshot()
  })
})
