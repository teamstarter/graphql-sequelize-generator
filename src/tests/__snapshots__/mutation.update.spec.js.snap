// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the create mutation Check that you can create a new instance of one entity 1`] = `
Array [
  "listGlobalBefore",
  "listBefore",
  "listAfter",
  "listBefore",
  "updateGlobalBefore",
  "updateBefore",
  "updateAfter",
  "listGlobalBefore",
  "listBefore",
  "listAfter",
  "listBefore",
]
`;

exports[`Test the create mutation Check that you can create a new instance of one entity: Updated user 1`] = `
Object {
  "__typename": "user",
  "id": 5,
  "name": "edited name",
}
`;

exports[`Test the create mutation Check that you can create a new instance of one entity: User 5 before update 1`] = `
Array [
  Object {
    "company": Object {
      "id": 1,
      "name": "Company 1",
    },
    "id": 5,
    "name": "Mr Test 5 c 2",
  },
]
`;

exports[`Test the create mutation Check that you can create a new instance of one entity: User 5 post update 1`] = `
Array [
  Object {
    "company": Object {
      "id": 1,
      "name": "Company 1",
    },
    "id": 5,
    "name": "edited name",
  },
]
`;

exports[`Test the create mutation Check that you can update an entity: Updated company 1`] = `
Object {
  "__typename": "company",
  "id": 5,
  "name": "updated name",
}
`;
